concordium-node/src/bin/cli.rs

```diff
@@ -399,15 +399,24 @@ fn instantiate_node(
     P2PNode::new(Some(node_id), conf, PeerType::Node, stats_export_service, regenesis_arc)
 }
 
+/// Establish initial connections to peers on the network.
+/// 1. Try connect to the provided "given_nodes" if configured.
+/// 2. Try connect to the peers that the node was last connected to (if any).
+/// 3. Try connect to the bootstrapper configured (if any) in order to advertise
+/// ourself such that the boostrapper node can provide us to freshly joining
+/// peers.
 fn establish_connections(conf: &config::Config, node: &Arc<P2PNode>) -> anyhow::Result<()> {
     info!("Starting the P2P layer");
     connect_to_config_nodes(node);
+    peers::connect_to_stored_nodes(node)?;
```

Comment:
lI don't think we should terminate here in case of error.

We should log the error instead and continue.